### DISCUSSION TITLE
setLayoutProperty, setFilter, setPaintProperty - does the order matter?

### DISCUSSION THREAD

**Discussion by pdebruic** [maplibre/maplibre-gl-js#2396] [Q&A] [answered]
To minimize the amount of work a browser has to do does the order of applying  `setLayoutProperty`, `setFilter`, or `setPaintProperty` matter?

I've got 3 settings a user could change to change what they see.  Sometimes changing all three of `setLayoutProperty`, `setFilter`, or `setPaintProperty` is necessary. 

Thanks

**Comment by HarelM** [comment]
I think `setFilter` is probably the most "important" as filtering features will reduce the amount of features, but if you call them one after the other I'm not sure (from javascript single threaded perspective) that it will make a difference.
But I'm not an expert of javascript performance...

**Reply by zhangyiatmicrosoft** [reply, marked as answer]
As far as I can tell, order does not matter.
And if you call them immediately one after another, not using setTimeout or requestAnimationFrame, they will be accomplished in the same render pass because of this "batching" logic:
https://github.com/maplibre/maplibre-gl-js/blob/c03607bad8a0d6574e9738b23fa84f3620713df5/src/ui/map.ts#L3060
